contracts/Example2.sol

```diff
@@ -1,11 +1,15 @@
+// SPDX-License-Identifier: MIT
 pragma solidity ^0.8.4;
 
 contract Example2 {
     uint public counter;
- 
-    function incrementBy(uint[] memory arr) public {
+
+    function incrementBy(uint[] calldata arr) external  {
+        uint _counter = counter;
+      //  uint8 length = uint8(arr.length);
         for (uint idx = 0; idx < arr.length; idx++){
```

Comment:
You can cache `arr.length` into a new variable (try using uint256 itself) and switch to `++i`.